src/Vault.sol

```diff
@@ -391,32 +390,46 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP, Multi
     /// @param vaultId Id of the vault
     /// @param amount The debt amount to be burned
     function burnDebt(uint256 vaultId, uint256 amount) external {
-        _updateVaultStabilisationFee(vaultId);
-
-        uint256 currentVaultDebt = vaultDebt[vaultId];
-        uint256 overallDebt = stabilisationFeeVaultSnapshot[vaultId] + currentVaultDebt;
-        amount = (amount < overallDebt) ? amount : overallDebt;
-        uint256 overallAmount = amount;
-
-        if (amount > currentVaultDebt) {
-            uint256 burningFeeAmount = amount - currentVaultDebt;
-            token.mint(treasury, burningFeeAmount);
-            stabilisationFeeVaultSnapshot[vaultId] -= burningFeeAmount;
-            amount -= burningFeeAmount;
+        uint256 currentNormalizationRate = _updateRateFee();
+
+        uint256 overallDebt = getOverallDebt(vaultId);
+        amount = (overallDebt < amount) ? overallDebt : amount;
+
+        token.transferFrom(msg.sender, address(this), amount);
+
+        globalDebt -= amount;
+        uint256 tokensToBurn;
+        uint256 currentMintsToBePaid = vaultMintedDebt[vaultId];
+
+        if (amount == overallDebt) {
+            tokensToBurn = currentMintsToBePaid;
+            vaultNormalizedDebt[vaultId] = 0;
+            vaultMintedDebt[vaultId] = 0;
+        } else {
+            tokensToBurn = FullMath.mulDiv(currentMintsToBePaid, amount, overallDebt);
+
+            uint256 currentNormalizedDebt = vaultNormalizedDebt[vaultId];
+            uint256 normalizedDebtToBurn = FullMath.mulDiv(amount, DEBT_DENOMINATOR, currentNormalizationRate);
```

Comment:
Can you research if using `mulDivRoundingUp` will result in correct accounting, without any division errors, so that all these if statements can be simplified?